src/main/java/sirius/db/mixing/query/QueryCompiler.java

```diff
@@ -344,7 +361,28 @@ private boolean continueToken(boolean inQuotes) {
                                effectiveMapping);
     }
 
-    private C compileConstraint(Tuple<Mapping, Property> mappingAndProperty, FieldValue token, boolean skipped) {
+    /**
+     * Compiles a constraint after a property has been resolved and an operator was detected.
+     * <p>
+     * This expects the tokenizer to look at an operator. The given <tt>mappingAndProperty</tt> will
+     * most probably contain a field/property which is addressed by the operation. In this case,
+     * an operation is created via {@link #parseOperation(Mapping, Property)}.
+     * <p>
+     * If the property cannot be resolved, the given <tt>token</tt> is passed into {@link #compileCustomField(String)}
+     * so that a sub class of the compiler can generate a constraint for a virtual field. If this also doesn't yield
+     * a constraint, a regular search in the default fields is generated. This might be necessarry so that tokens like
+     * an:value can be used as search term as long as no property named "an" exists.
+     *
+     * @param mappingAndProperty the mapping (join path) and the resolved property to address
+     * @param token              the token which has been parsed as field / property name
+     * @param skipped            <tt>true</tt> if we previously skilled whitespaces. Therefor fore "x=y" we generate
+     *                           this as literal term. But for "x =y" we simply generate a search for "x" and leave
```

Comment:
```suggestion
     *                           this as literal term. But for "x=y" we simply generate a search for "x" and leave
```

Reply:
Nope, this whitespace is put there on purpose... as "x =y" and "x=y" ist tokenized differently as long as x isn't a field / property.